docs/release.rst

```diff
@@ -1521,11 +1521,11 @@ Bug fixes
 Documentation
 ~~~~~~~~~~~~~
 
-* Some changes have been made to the :ref:`spec_v2` document to clarify
+* Some changes have been made to the Zarr Specification v2 document to clarify
```

Comment:
Let's make this a hyperlink to the zarr-specs page

Reply:
Sure, once we merge https://github.com/zarr-developers/zarr-specs/pull/276, I'll add the link here.